### PR TITLE
feat: split schematics component analytics

### PR DESCRIPTION
**Pull request by matthieu-crouzet**
## Proposed change
Split of the analytics part of the schematic to generate a component

### CODE REVIEW

File: packages/@o3r/analytics/schematics/analytics-to-component/schema.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import type { JsonObject } from '@angular-devkit/core';
+
+export interface NgAddAnalyticsSchematicsSchema extends JsonObject {
+  /** Path to the component */
+  path: string;
+
+  /** Skip the linter process */
+  skipLinter: boolean;
+
+  /** Determine if the dummy localization should be generated */
```
Comment (mrednic-1A) [inline packages/@o3r/analytics/schematics/analytics-to-component/schema.ts:10]: it's dummy inputs for analytics, not localization :)

**Review by vscaiceanu-1a — DISMISSED** [review]

File: packages/@o3r/core/schematics/component/common/analytics.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        context.logger.info(`
+          You can add analytics later to this component via this command:
+          ng g ${analyticsPackageName}:${addAnalyticsSchematicName} --path="${componentPath}"
+        `);
+      }
+    }
+  } catch {
+    if (useOtterAnalytics) {
+      throw new Error(`
+        You need to install '${analyticsPackageName}' to be able to generate component with configuration.
```
Comment (cpaulve-1A) [inline packages/@o3r/core/schematics/component/common/analytics.ts:48]:
```suggestion
        You need to install '${analyticsPackageName}' to be able to generate component with analytics.
```

**Review by vscaiceanu-1a — DISMISSED** [review]

File: packages/@o3r/analytics/schematics/analytics-to-component/schema.json [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    "skipLinter": {
+      "type": "boolean",
+      "description": "Skip the linter process",
+      "default": false
+    },
+    "activateDummy": {
+      "type": "boolean",
+      "description": "Generate dummy values",
+      "default": false,
+      "x-prompt": "Generate dummy analytics event"
```
Comment (kpanot) [inline packages/@o3r/analytics/schematics/analytics-to-component/schema.json:20]: do we want to keep this prompt?

File: packages/@o3r/analytics/schematics/analytics-to-component/schema.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+export interface NgAddAnalyticsSchematicsSchema extends JsonObject {
+  /** Path to the component */
+  path: string;
+
+  /** Skip the linter process */
+  skipLinter: boolean;
+
+  /** Determine if the dummy analytics events should be generated */
+  activateDummy: boolean | null;
```
Comment (kpanot) [inline packages/@o3r/analytics/schematics/analytics-to-component/schema.ts:11]: it is not `nullable`

File: packages/@o3r/core/schematics/component/common/analytics.ts [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+  context: SchematicContext
+): Promise<Rule[]> => {
+  let useOtterAnalytics = options.useOtterAnalytics;
+  let alwaysUseOtterAnalytics;
+  try {
+    require.resolve(`${analyticsPackageName}/package.json`);
+    if (useOtterAnalytics === null && context.interactive) {
+      useOtterAnalytics = await askConfirmation('Generate component with Otter analytics?', true);
+      if (useOtterAnalytics) {
+        alwaysUseOtterAnalytics = await askQuestion('Generate future components with Otter analytics by default?', [
```
Comment (kpanot) [inline packages/@o3r/core/schematics/component/common/analytics.ts:21]: Could make sense to have it shared in the `@o3r/schematics` package with a more generic question
(such as "Use this response as default value?")

File: tsconfig.base.json [hunk 6]
```diff
@@ -65,9 +65,6 @@
       "@o3r/analytics": [
         "packages/@o3r/analytics/src/public_api"
       ],
-      "@o3r/analytics/*": [
```
Comment (kpanot) [inline tsconfig.base.json:68]: Needed for the `plugins` subentrie, can be replaced by `@o3r/analytics/plugins/*` if it is fixing your issue

File: packages/@o3r/analytics/schematics/analytics-to-component/schema.json (hunk 3 above)
Comment (matthieu-crouzet) [reply packages/@o3r/analytics/schematics/analytics-to-component/schema.json:20]: I keep this prompt for others schematics 
maybe I can remove it in another PR, to keep it all aligned
(I will do another PR to refactor a bit all the component schematics code to share a maximum of lines, that we didn't think we can share at the beginning)

File: packages/@o3r/core/schematics/component/common/analytics.ts (hunk 5 above)
Comment (matthieu-crouzet) [reply packages/@o3r/core/schematics/component/common/analytics.ts:21]: I will do another PR to refactor a bit all the component schematics code to share a maximum of lines, that we didn't think we can share at the beginning
I can include this fix in this PR

File: packages/@o3r/analytics/schematics/analytics-to-component/schema.json (hunk 3 above)
Comment (kpanot) [reply packages/@o3r/analytics/schematics/analytics-to-component/schema.json:20]: Ok then, let's remove it in another PR

**Review by mrednic-1A — APPROVED** [review]

**Review by vscaiceanu-1a — APPROVED** [review]

**Pull request merged**